samples/nrf_desktop/src/hw_interface/motion_optical.c

```diff
@@ -744,13 +810,47 @@ static void optical_thread_fn(void)
 				cpi = new_cpi;
 				update_cpi(cpi);
 			}
+
+			u32_t new_time = atomic_get(&sensor_downshift_run);
+			if (new_time != time_run) {
+				time_run = new_time;
+				update_downshift_time(
+					OPTICAL_REG_RUN_DOWNSHIFT,
+					time_run);
+			}
+
+			new_time = atomic_get(&sensor_downshift_rest1);
+			if (new_time != time_rest1) {
+				time_rest1 = new_time;
+				update_downshift_time(
+					OPTICAL_REG_REST1_DOWNSHIFT,
+					time_rest1);
+			}
+
+			new_time = atomic_get(&sensor_downshift_rest2);
+			if (new_time != time_rest2) {
+				time_rest2 = new_time;
+				update_downshift_time(
+					OPTICAL_REG_REST2_DOWNSHIFT,
+					time_rest2);
+			}
 		}
 
 	}
 	/* This thread is supposed to run forever. */
 	__ASSERT_NO_MSG(false);
 }
 
+static void set_downshift_time_from_bytes(atomic_t *sensor_val,
+					  const u8_t *data)
+{
+	/* Convert little endian bytes to an unsigned integer. */
+	u32_t new_time = data[0] + (data[1] << 8)
```

Comment:
I think we have utility function that does the conversion for us (check hids)

Reply:
`sys_get_le16()` and `sys_get_le32()` look useful, thanks!